option.go

```diff
@@ -149,8 +149,49 @@ func panicsIfNotCorrectType(openapiParam *openapi3.Parameter, exampleValue any)
 	return exampleValue
 }
 
-// Registers a parameter for the route. Prefer using the [Query], [QueryInt], [Header], [Cookie] shortcuts.
-func OptionParam(name string, options ...func(*OpenAPIParam)) func(*BaseRoute) {
+// Declare a response header for the route.
+// This will be added to the OpenAPI spec, under the 200 response.
+// Example:
+//
+//	ResponseHeader("Content-Range", "Pagination range", ParamExample("42 pets", "unit 0-9/42"), ParamDescription("https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range"))
+//	ResponseHeader("Set-Cookie", "Session cookie", ParamExample("session abc123", "session=abc123; Expires=Wed, 09 Jun 2021 10:18:14 GMT"))
+//
+// The list of options is in the param package.
+func OptionResponseHeader(name, description string, options ...func(*OpenAPIParam)) func(*BaseRoute) {
+	apiParam, openapiParam := buildParam(name, options...)
+
+	openapiParam.Name = ""
+	openapiParam.In = ""
+
+	if len(apiParam.StatusCodes) == 0 {
+		apiParam.StatusCodes = []int{200}
+	}
+
+	return func(r *BaseRoute) {
+		for _, code := range apiParam.StatusCodes {
+			codeString := strconv.Itoa(code)
+			response200 := r.Operation.Responses.Value(codeString)
```

Comment:
The variable name is confusing for me here. IS the only available for 20x error codes?

If so maybe name this `response2XX`. Also maybe we should validate that the user is only providing 200s?

Reply:
The naming will be changed after merging the other PR

Reply:
Ready for review again :)